cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/SparkCqlEvaluatorTest.java

```diff
@@ -125,7 +129,132 @@ public void testAllTypesEvaluationSuccess() throws Exception {
+                "--input-format", "parquet",
+                "-i", "Type1=" + new File(inputDir, "testdata/Type1").toURI().toString(),
+                "-i", "Type2=" + new File(inputDir, "testdata/Type2").toURI().toString(),
+                "-o", "Context1Id=" + context1IdFile.toURI().toString(),
+                "-o", "Context2Id=" + context2IdFile.toURI().toString(),
+                "--overwrite-output-for-contexts"
+        };
+
+        SparkCqlEvaluator.main(args);
+        
+        
+        validateOutput(
+                context1IdFile.toURI().toString(),
+                Arrays.asList(
+                        RowFactory.create(0, null, null, null, false),
+                        RowFactory.create(1, null, null, null, false),
+                        RowFactory.create(2, 33, "string1", new BigDecimal(9.989), true),
+                        RowFactory.create(3, 22, "string2", new BigDecimal(-2.816), true),
+                        RowFactory.create(4, 22, "string1", new BigDecimal(-4.926), true)
+                ),
+                new StructType()
+                        .add("id", DataTypes.IntegerType, false)
+                        .add("Context1Id.define_integer", DataTypes.IntegerType, true)
+                        .add("Context1Id.define_string", DataTypes.StringType, true)
+                        // Decimal precision currently hardcoded in QNameToDataTypeConverter
+                        .add("Context1Id.define_decimal", DataTypes.createDecimalType(28, 8), true)
+                        .add("Context1Id.define_boolean", DataTypes.BooleanType, true)
+        );
+
+        validateOutput(
+                context2IdFile.toURI().toString(),
+                Arrays.asList(
+                        // Expected Instants derived from reading the input files and printing out
+                        // the datetime column using UTC
+                        RowFactory.create(0, LocalDate.of(2002, 9, 27), Instant.parse("2002-05-31T13:19:08.512Z")),
+                        RowFactory.create(1, LocalDate.of(2002, 3, 10), Instant.parse("2002-05-07T09:19:19.31Z")),
+                        RowFactory.create(2, LocalDate.of(2002, 5, 29), Instant.parse("2002-03-13T13:00:52.859Z")),
+                        RowFactory.create(3, null, null),
+                        RowFactory.create(4, LocalDate.of(2002, 5, 21), Instant.parse("2002-02-17T08:59:43.793Z"))
+                ),
+                new StructType()
+                        .add("id", DataTypes.IntegerType, false)
+                        .add("Context2Id.define_date", DataTypes.DateType, true)
+                        .add("Context2Id.define_datetime", DataTypes.TimestampType, true)
+        );
+    }
+    
+    private void validateOutput(String filename, List<Row> expectedRows, StructType schema) {
+        spark = initializeSession(Java8API.ENABLED);
+        Dataset<Row> actualDataFrame = spark.read().parquet(filename);
```

Comment:
If we add an `--output-format` option to the main program, we will need to be able to control for that here as well.